### DISCUSSION TITLE
How to generate .cert and .key files for XMSS and LMS algorithm using oqs-provider?

### DISCUSSION THREAD

**Discussion by TheGodOfObelisk** [open-quantum-safe/oqs-provider#471] [Q&A]
I hope to use oqs-provider to generate key and certificates for XMSS and LMS family algorithms. I know that the repository "liboqs" supports these algorithms by setting a configuration option to "ON". Is it possible to realize this in "oqs-provider"? For example, how can I generate the key and certificate of XMSS use a command like the following one.
`openssl req -x509 -sha256 -newkey [XMSS] -keyout XMSS.key -out XMSS.crt -nodes -days 365`

Thank you very much!
Best regards

**Comment by baentsch** [comment]
Not yet (implemented): See #331. PR welcome.

**Reply by baentsch** [reply]
Ooops -- I just re-thought what's "behind" your ask, @TheGodOfObelisk 

> I hope to use oqs-provider to generate key and certificates for XMSS and LMS family algorithms

And the answer to your question

> Is it possible to realize this in "oqs-provider"? 

changes to Never as I share the [sentiment expressed here](https://github.com/open-quantum-safe/liboqs/pull/1650#issuecomment-1893957110). 

What may become possible if someone volunteers the work is signature verification -- basically anything that can be implemented without activating [OQS_HAZARDOUS_EXPERIMENTAL_ENABLE_SIG_STFL_KEY_SIG_GEN](https://github.com/open-quantum-safe/liboqs/blob/main/CONFIGURE.md#stateful-hash-based-signatures).
